internal/store/user/user.go

```diff
@@ -7,11 +7,18 @@ type User struct {
 	IsAdmin        bool   `json:"isAdmin"`
 }
 
+// NullableUser is a nullable version of user.
+type NullableUser struct {
+	Username       *string `json:"username"`
+	HashedPassword *string `json:"hashedPassword"`
+	IsAdmin        *bool   `json:"isAdmin"`
+}
+
 // Service is a store for users.
 type Service interface {
 	Create(User) error
 	Get(username string) (User, error)
 	GetUsers() ([]User, error)
-	Update(username string, u User) error
+	Update(username string, nu NullableUser) error
```

Comment:
Do we need to supply the username separately from the user data?

Reply:
Yes for changing a username.

Reply:
Oh duh 👍